src/packet.rs

```diff
@@ -378,9 +378,11 @@ impl Blob {
                     }
                     return Err(Error::IO(e));
                 }
-                Ok(()) => if i == 0 {
-                    socket.set_nonblocking(true)?;
-                },
+                Ok(()) => {
```

Comment:
next time create a completely different PR for formatting changes, and just land it.  Nobody needs to review this kind of stuff, and it wastes mental energy that would otherwise be looking at the real changes